src/main/java/org/jenkinsci/plugins/mesos/MesosApi.java

```diff
@@ -64,36 +64,43 @@
    * @param jenkinsUrl The Jenkins address to fetch the agent jar from.
    * @param agentUser The username used for executing Mesos tasks.
    * @param frameworkName The name of the framework the Mesos client should register as.
+   * @param frameworkId The id of the framework the Mesos client should register for.
    * @param role The Mesos role to assume.
    * @throws InterruptedException
    * @throws ExecutionException
    */
   public MesosApi(
-      URL masterUrl, URL jenkinsUrl, String agentUser, String frameworkName, String role)
+      URL masterUrl,
+      URL jenkinsUrl,
+      String agentUser,
+      String frameworkName,
+      String frameworkId,
+      String role)
       throws InterruptedException, ExecutionException {
     this.frameworkName = frameworkName;
-    this.frameworkId = UUID.randomUUID().toString();
+    this.frameworkId = frameworkId;
```

Comment:
USI should do this

Reply:
How should USI persist the framework id? We don't have any persistence layer yet. I don't think this is in scope for running Jenkins on MWT.

Reply:
Oh, maybe I misunderstood. You mean USI should define the framework id? I agree. However, I'm a little uncertain how this will be done. `MesosCloud` needs to know the id to persist it.

Reply:
@timcharper, as far as I can tell USI does not generate the framework id yet. Should we do a follow up? I don't think it should block this PR.